core/core.js

```diff
@@ -9,7 +9,7 @@
   let opsCache = {};
   const errorMap = {};
   let nextPromiseId = 1;
-  const promiseTable = new Map();
+  const promiseTable = Object.create(null);
```

Comment:
CC @AaronO my flamegraphs shown that using `Map` still causes significant performance problems on the GC side; changing it back to plain object gives about 5-10% throughput increase

Reply:
> @bartlomieju Do you mind if we split that out into a separate PR ?
> 
> Because there are some ideas I want to investigate, potentially something faster than `Map` or `Object` (reducing GC-pressure).

Not at all, interested to hear your ideas